packages/mermaid/src/diagrams/common/common.ts

```diff
@@ -208,21 +208,33 @@ export const parseGenericTypes = function (input: string): string {
   return output.join('');
 };
 
+export const countOccurrence = (string: string, substring: string): number => {
```

Comment:
If we're `export`-ing this function from the file, it would be helpful to have a TSDoc comment, even if it's super basic and a bit repetitive for such a simple function:

```suggestion
/**
 * Count the number of times the given substring appears in the string.
 *
 * @param string - The input string.
 * @param substring - The string to search for.
 * @returns The number of times the substring appears in the string.
 */
export const countOccurrence = (string: string, substring: string): number => {
```

At some point in the future, we could add an ESLint rule like `"jsdoc/require-jsdoc": ["warn", {"publicOnly":true`}]` to warn us about exported functions that aren't documented.

Reply:
I prefer functions that are self explanatory. Comments are useful for explaining complex steps and functions, but seems like an unnecessary addition for simple functions.

Reply:
> Comments are useful for explaining complex steps and functions, but seems like an unnecessary addition for simple functions.

I agree! But, I do think if we're exporting a function out of a file, the bar for documenting it is a lot lower, since you can't easily see the source-code of the function (especially if you're minifying code).

I think the name/params for this function are self-explanatory enough that we don't need a TSDoc comment, and it'll only be necessary if we ever add an ESLint rule like `"jsdoc/require-jsdoc": ["warn", {"publicOnly":true}]`.